### ISSUE TITLE
bypass 330Mhz clock limit?

### ISSUE THREAD

**Issue by shad00m** [Floris497/mac-pixel-clock-patch-V2#178]
I'm trying to overclock my Qnix QX2710 but I only get to around 80hz because after that the pixel clock reaches 330Mhz. Anything above that does not work in switchresx. Would anyone know if theres a bypass for that?

**Comment by Floris497** [comment]
You gotta give me more info about your setup..

**Comment by shad00m** [comment]
Sorry.
Hackintosh macOS 10.12
Intel Xeon 1230v2
Nvidia GTX970
Monitor: QNIX QX2710 (can do about 110Hz in windows)
SwitchresX will not "install" the custom resolutions that go beyond 330Mhz

**Comment by Floris497** [comment]
And what cable do you use?

**Comment by shad00m** [comment]
Its a DVI-DL

**Comment by shad00m** [comment]
I'm running 386 Mhz on Windows though. There are unlockers for windows but not all cards need those unlockers. (https://www.monitortests.com/forum/Thread-NVIDIA-Pixel-Clock-Patcher)
I thought your program has the same purpose but it appears I was mistaken. Sorry

**Comment by Floris497** [comment]
> Dual-Link DVI, adds a 3 additional differential pairs over which to transfer RGB. The clock link is shared between the 2 sets, so now you have 7 instead of 4 wires to transfer image data. The original set of regular DVI is used to transfer odd pixels, the extra pair is used to transfer even pixels. The DVI specification says that DL-DVI must be used for pixel clocks that go above 165 MHz. A common mistake is to assume that DL-DVI is thus limited to a pixel clock of 330 MHz, since there are now simply 2 pixels in parallel. This is not the case: there is no such limitation and sources and sinks are allowed to jack up the pixel clock to whatever rate they want. Note that even the Wikipedia on DL-DVI makes this mistake! That doesn't mean that there aren't any limitations: older GPUs weren't designed to transfer pixels at pixel rates that were higher than 330 MHz, so that automatically became the limit. A good example is 2560 x 1440 @ 120 Hz: this is a video mode that is routinely achieved by some direct drive monitors over DL-DVI. When using CVT2-RB timings, this requires a clock speed of 483 MHz, far higher than incorrect 330 MHz limit.
\- http://monitorinsider.com/cable_bandwidth.html


So it looks like some GPU's can do more than 330MHz.. did you try the CoreDisplay Patch?

**Comment by shad00m** [comment]
Yeah I've tried that. It would let me go beyond the standard 60Hz but switchresx would not enable any resolution above Pixelclock 330Mhz
![screen shot 2017-05-13 at 16 29 00](https://cloud.githubusercontent.com/assets/9115740/26026313/4fc2699c-37f9-11e7-9fb3-10d71ff3f9ea.png)

**Comment by Floris497** [comment]
Hm.. same for me, my DL-DVI does not go beyond 330MHz let's have some fun with the Nvidia driver, I'll let you know what I find :)

**Comment by shad00m** [comment]
Thank you very much @Floris497

**Comment by Floris497** [comment]
So I've got you some good new and some bad news:
- I've got it working..
- but on pascal

Means that it will be fine on Maxwell as well, but just have to do the calculations :)
And I have to think of a way to build this into 1 script.. for different architectures..

I'll post a command here that you can test with if you want?

**Comment by shad00m** [comment]
Sure! I'll be happy to help.

**Comment by Floris497** [comment]
`perl -i.bak -pe '$oldLimit1 = qr"\x8B\x82\xB0\x00\x00\x00\xB9\x88\x84\x02\x00"s;$newLimit1 = "\x8B\x82\xB0\x00\x00\x00\xB9\x00\x35\x0C\x00";s/$oldLimit1/$newLimit1/g' /System/Library/Extensions/NVDAGM100HalWeb.kext/Contents/MacOS/NVDAGM100HalWeb`

**Comment by Floris497** [comment]
This should do the trick :)

**Comment by shad00m** [comment]
Do I need to do anything else besides that? It didn't seem to change anything.
Resolutions stayed "Not Installed" and when I try them with "Apply Now (experimental)" the screen turns upside down and it writes "The Display configuration has been modified"

**Comment by Floris497** [comment]
Ehm.. no you have to keep both patches.. don't use the experimental thing.. just save and reboot :) Do you have another PC or phone where you can remote in to the mac? because sometimes when you patch switchresx or macOS will set a resolution that is not supported by the screen.. making it very difficult to change it..

**Comment by shad00m** [comment]
Yeah, I do have remote access to it.
But it persists that a resolution with 329.99Mhz stays "Active" and 330.32 as "Not Installed"

**Comment by Floris497** [comment]
hm.. I do not have that button.. what was the output of that piece of code you ran? did it say anything?

**Comment by shad00m** [comment]
nope no output at all

**Comment by Floris497** [comment]
In the custom resolution window use 'CMD + S' to save than the resolutions should say something like 'needs reboot' after this reboot

**Comment by Floris497** [comment]
Wait.... what version of WEB drivers are you on? I'm on "378.05.05.05f02"

**Comment by Floris497** [comment]
md5 /System/Library/Extensions/NVDAGM100HalWeb.kext/Contents/MacOS/NVDAGM100HalWeb should be: `354f95268a7f5a24106810ed18d951d5` after patching

**Comment by shad00m** [comment]
Its d7c5490ec995733c57f0ea53b94bb90f for me
I'm on 378.05.05.05f02. Should I reinstall the driver and patch again?

**Comment by Floris497** [comment]
That is correct.. my bad

**Comment by shad00m** [comment]
https://youtu.be/EJzXWx1LfY4
Pixelclock:
320 Mhz => Save => Needs Reboot
335 Mhz => Save => Not Installed

Maybe theres a limit in switchresx that stops me from doing that?

**Comment by Floris497** [comment]
![screen shot 2017-05-13 at 19 57 41](https://cloud.githubusercontent.com/assets/5049111/26027895/97719da4-3816-11e7-8c88-bf3f86d9cac3.png)

**Comment by shad00m** [comment]
I'm such a silly goose. I totally forgot about the 2 Custom Resolution limit for free users... It totally works now! Thank you for your help thats great!

**Comment by Floris497** [comment]
:D enjoy

**Comment by shad00m** [comment]
@Floris497 I'm sorry to bug you again but did this stop working with the new macOS update?
I installed the webdrivers - ran coredisplay.command - ran your perl command but it wont work again :( 

MD5 (/System/Library/Extensions/NVDAGM100HalWeb.kext/Contents/MacOS/NVDAGM100HalWeb) = 059e81a8ffb42e41e57de6686a71d6e4

**Comment by Floris497** [comment]
Hm, I'm actually not sure.. but it should run fine on 10.12.5 although I think I had it on 10.12.4.. I'll check tomorrow everything is shut down right now.. could you see if MD5 of patched and I patched driver are different? And make sure you are running web drivers and not normal drivers :)

**Comment by shad00m** [comment]
Web driver 15f01: e351c25f7f11665b4c8b38675b7d3a51
After patching: e351c25f7f11665b4c8b38675b7d3a51

So I guess they changed something on the upgrade from 05f02 to 15f01 in the Webdriver

**Comment by Floris497** [comment]
Hm.. I'll have a look tomorrow morning.. so in about 8-10 hours :) but it should not have changed.. did you get any feedback while patchung nvidia driver?

**Comment by shad00m** [comment]
This is odd... I ran the perl command another time and now the md5 changed back to what is was before. I must have typed my password wrong or forgot the sudo.
New md5 is: 059e81a8ffb42e41e57de6686a71d6e4
Which is the same md5 I had before and it didn't work... now it does though. I guess its too late for me... Sorry for disturbing :(

**Comment by Floris497** [comment]
Not a problem at all :) enjoy!

**Comment by passmoregames** [comment]
Hey Floris - I realise I am resurrecting something here, so apologies for that.

I am running a hackintosh with 10.13.3 + Nvidia 780GTX.
I too have a QNIX 2710, and i am able to overclock it way past 330mhz in windows, but in mac i seem to be stuck at this limit. I have tried your coredisplay patch, and i have also tried your maxwell patch - though i am pretty sure my card is keplar. I have the same problem that the other guy was getting before you fixed his problem - anything above 330mhz gets the 'not activated - invalid?' message on switchresx. 

Do you think you would be able to help me? 

Thanks a lot,
Dave.

**Comment by Floris497** [comment]
Please ask me again to look at it if I don't respond more detailed by the end of tomorrow

**Comment by passmoregames** [comment]
Hey Floris

It looks like i did a terrible job of reminding you to look at this again, sorry about that! 

Any thoughts?

Best,
Dave.

**Comment by Floris497** [comment]
Right.. that was like a small 17 hours times 24

I'll give it a look when i get home :)

**Comment by Floris497** [comment]
It'll be tomorrow.. my apologies..

**Comment by passmoregames** [comment]
No problem at all! Did you have any luck?

**Comment by Floris497** [comment]
Now it's becoming a problem..... I'm so sorry.. i just don't have a lot of time to check it..

**Comment by passmoregames** [comment]
Seriously, nothing to apologise for. Let me know if you have the time at some point, if not, no problem.
